pkg/relayer/proxy/sync.go

```diff
@@ -0,0 +1,222 @@
+package proxy
```

Comment:
Note: i didn't re-review this file assuming it's the same as what we had previously.